data_merger.py

```diff
@@ -138,6 +141,22 @@ def add_leftover_datapass_and_remove_matched_rows(self, datapass_content):
             datapass_content.drop(datapass_row.name, inplace=True)
 
         return output_rows
+    
+    def last_event_is_refuse(self, events):
+        if not events:
+            return False
+        last_event = max(events, key=lambda x: x["created_at"])
+        return last_event.get("name") == "refuse"
+
+    def add_reopened_and_refused_datapass_and_remove_matched_rows(self, datapass_content):
+        output_rows = []
+        for _, datapass_row in datapass_content.iterrows():
+            if datapass_row["Statut"] == "Accepté" and self.last_event_is_refuse(datapass_row["Events"]):
```

Comment:
Cette condition est ok pour trouver les refus récents. de réouvertures.

Mais si le refus date d'il y a quelque mois, et que entre temps une autre réouverture a été acceptée, on ne va pas pouvoir rentrer dans ce cas pour matcher la ligne.

Et si plusieurs refus existent dans l'historique, on ne va pas tous les considérer non plus.

Reply:
Ah mais en plus non, la condition ne marchera même pas, par ce que tu n'auras pas de row dans le datapass content qui existe avec cette condition.

Reply:
En fait, l'algo c'est plutôt d'itérer sur le reste d'**input content**, et de vérifier si les lignes de ce reste ne correspondent pas à des refus de réouverture.

- Si c'est des lignes qui ont déjà le statut de refus, on les rajoute à l'output telles quelles
- Si c'est des lignes qui sont en cours d'instruction de réouverture, et qu'on arrive à vérifier qu'elles ont été refusées dans les données de datapass, alors on les update en "Refusé" et on les rajoute à l'output